extensions/harvest/src/services/harvest.ts

```diff
@@ -93,6 +93,8 @@ export function useActiveClients() {
 }
 
 export function useMyProjects() {
+  const cachedProjects = JSON.parse(cache.get("projects") ?? "[]") as HarvestProjectAssignment[];
```

Comment:
This feels like an extra, unnecessary step since I'm already using the "useCachedPromise" function, but it seems to fix the issues... Please correct me if there is a better way that I'm missing

Reply:
Are you able to make a very simple repro case since I don't have this extension or knows how it works 🙂

Reply:
Yeah I use this extension every day and this fix does work, I just don’t understand why. I feel like I may be using the use cached promise function incorrectly. Or misunderstand it

Let me know if you need a screencast

Reply:
I'd like to view a simple screencast of the issue so I can discuss it internally :)

Reply:
https://github.com/raycast/extensions/assets/37158449/85a45e5d-b79a-4960-8c91-901e79902d18

Reply:
If the cache doesn't work, it means that the hash of the function changes (and so the hooks thinks it's a new function.

Can you try it as 

```js
async function fetchProjects() {
  let project_assignments: HarvestProjectAssignment[] = [];
      let page = 1;
      // eslint-disable-next-line no-constant-condition
      while (true) {
        const resp = await harvestAPI<HarvestProjectAssignmentsResponse>({
          url: "/users/me/project_assignments",
          params: { page },
        });
        project_assignments = project_assignments.concat(resp.data.project_assignments);
        if (resp.data.total_pages >= resp.data.page) break;
        page += 1;
      }
      return project_assignments;
}
export function useMyProjects() {
  return useCachedPromise(
    fetchProjects,
    [],
    { initialData: [] }
  );
}
```